tests/test_agpypeline.py

```diff
@@ -0,0 +1,487 @@
+
+TEST_IMAGE = os.path.realpath('images/output.tin.tif')
+
+TEST_METADATA = os.path.realpath('data/08f445ef-b8f9-421a-acf1-8b8c206c1bb8_metadata_cleaned.json')
+
+
+def test_exists():
+    """Tests if the files stored in variables above are accessible/contained"""
+    for file in TEST_FILES:
+        assert os.path.isfile(file)
+    assert os.path.isfile(TEST_IMAGE) and os.path.isfile(TEST_METADATA)
+
+
+# Other methods can be used here if wanted in order to test if the error is raised, but this would
+# require creating a class
+def test_algorithm():
+    """Tests initializing the algorithm.Algorithm() class within agpypeline and performing a process with default
+    parameters """
+    alg = algorithm.Algorithm()
+    try:
+        alg.perform_process(environment.Environment(configuration.Configuration()), {}, {}, [])
+        assert True is False
+    except RuntimeError as error:
+        assert str(error) == "The Algorithm class method perform_process() must be overridden by a derived class"
+
+
+def test_configuration():
+    """Tests agpypeline's configuration.Configuration() class by initializing configuration.Configuration() and
+     checking to make sure that all necessary parameters are contained"""
+    config = configuration.Configuration
+    for entry in ['transformer_version', 'transformer_description', 'transformer_name', 'transformer_sensor',
+                  'transformer_type', 'author_name', 'author_email', 'contributors', 'repository']:
+        assert hasattr(config, entry)
+        if entry == 'contributors':
+            assert getattr(config, entry) == []
+        else:
+            assert getattr(config, entry) is None
+
+
+def test_entrypoint_handle_error():
+    """Tests entrypoint's handle_error function by passing in an error code and message"""
+    entry = entrypoint.__internal__()
+    bad_result = entry.handle_error(None, None)
+    assert bad_result == {'error': 'An error has occurred with error code (-1)', 'code': -1}
+    result = entry.handle_error(0, "test message")
+    assert result == {'error': 'test message', 'code': 0}
+
+
+def test_entrypoint_load_metadata():
+    """Tests entrypoint's load_metadata function by seeing if the result is the same as a previously called valid
+    result"""
+    entry = entrypoint.__internal__()
+    orig = json.load(open("data/entrypoint_load_metadata_orig.json"))
+    bad_result = entry.load_metadata("")
+    assert bad_result == {'error': "Unable to load metadata file ''"}
+    assert entry.load_metadata("data/08f445ef-b8f9-421a-acf1-8b8c206c1bb8_metadata_cleaned.json") == orig
+
+
+def test_entrypoint_check_params_result_error():
+    """Tests entrypoint's check_params_result_error function by passing in both an entry dictionary and
+    a dictionary with a simple error message and code and checking for correct output"""
+    entry = entrypoint.__internal__()
+    result_none = entry.check_params_result_error({})
```

Comment:
The inputs and results could be put into a JSON file. It makes it easier to extend the tests